### PR TITLE
165 fix login bug

### PR DESCRIPTION
**Pull request by TheodoreRamey**

### CODE REVIEW

**Comment by TheodoreRamey** [comment]
Connor and I tested this manually via PostMan to confirm that both invalid and valid credentials returned the LoginResponse as expected (401 for invalid and 200 with a jwt for valid)

**Comment by c-lawrence** [comment]
I can confirm we got the login authentication working in PostMan. Want to see if we can do Unit Testing for this before merging, but need a bit more research to get automated testing working for out particular project.

File: back-end/src/main/kotlin/Gneiss/PacketCompiler/Service/Login.kt [hunk 1]
```diff
@@ -9,21 +9,25 @@ import org.springframework.http.ResponseEntity
...
 
         // If validCredentials is false, return a 401 response 'authentication failed'
         // Otherwise return a 200 response with a jwt token as the body
         if (!validCredentials.validFlag) {
-            return ResponseEntity<String>("Invalid Credentials", HttpStatus.UNAUTHORIZED)
+            return ResponseEntity<LoginResponse>(LoginResponse("Invalid Credentials"), HttpStatus.UNAUTHORIZED)
         } else {
             val jwt = jwtHelper.createJWT(JWTBody(req.username, validCredentials.roleId))
-            return ResponseEntity<String>(jwt, HttpStatus.OK)
+            println(jwt)
```
Comment (jacksontheel) [inline back-end/src/main/kotlin/Gneiss/PacketCompiler/Service/Login.kt:28]: Did you mean to leave this in?

File: back-end/src/main/kotlin/Gneiss/PacketCompiler/DatabaseAccess/UserDao.kt [hunk 2]
```diff
@@ -35,25 +35,26 @@ class UserDao {
...
         // If the credentials are valid, we need to get the role_id out of the result
         // To do this, we reset the result set pointer to the first row with beforeFirst() and next()
         // And then get the int in the fourth position, where role_id is the query
         if (rowCount == 1) {
             resultSet.beforeFirst()
             resultSet.next()
-            val roleId = resultSet.getInt(4)
-            return CredentialsResponse(true, roleId.toString())
+            val roleId = resultSet.getString(4)
+            connection!!.close()
```
Comment (jacksontheel) [inline back-end/src/main/kotlin/Gneiss/PacketCompiler/DatabaseAccess/UserDao.kt:53]: Normally for closing resources, a finally block, or Kotlin's [use](https://www.baeldung.com/kotlin/try-with-resources) function is best, but I don't see much of a risk of anything here throwing, so I think this is fine.

Another benefit of using one of the strategies above is knowing at compile time if we're going to be using a resource that's been closed.

**Review by jacksontheel — APPROVED** [review]
Looks awesome. You may have left a println in there by mistake.

**Pull request merged**